src/LoansNFT.sol

```diff
@@ -482,41 +512,36 @@ contract LoansNFT is ILoansNFT, BaseNFT {
         returns (uint takerId, uint providerId, uint loanAmount)
     {
         require(configHub.canOpen(address(takerNFT)), "unsupported taker contract");
-        // provider contract is valid
+        // provider contract is set
         require(address(currentProviderNFT) != UNSET, "provider contract unset");
         // 0 underlying is later checked to mean non-existing loan, also prevents div-zero
         require(underlyingAmount != 0, "invalid underlying amount");
 
         // swap underlying
         // @dev Reentrancy assumption: no user state writes or reads BEFORE the swapper call in _swap.
-        // The only state reads before are owner-set state: pause and swapper allowlist.
-        uint cashFromSwap = _swapUnderlyingWithTwapCheck(underlyingAmount, swapParams);
-
-        uint putStrikePercent = currentProviderNFT.getOffer(offerId).putStrikePercent;
+        // The only state reads before are owner-set state (e.g., pause and swapper allowlist).
+        uint cashFromSwap = _swap(underlying, cashAsset, underlyingAmount, swapParams);
+        /* @dev note on swap price manipulation:
```

Comment:
this note "replaces" the now removed check of swap vs. twap price (on open). The review note:

> - #low MAX_SWAP_TWAP_DEVIATION_BIPS is too low, and is a DoS risk => raise to 50%, since only for opens and is only a sanity check. Consider removing completely since benefit is unclear, but adds complexity and dos vectors and scenarios (e.g., volatility)